### ISSUE TITLE
[question] Deprecation of XUL and XPCOM

### ISSUE THREAD

**Issue by haasn** [5digits/dactyl#82]
https://blog.mozilla.org/addons/2015/08/21/the-future-of-developing-firefox-add-ons/
With the recent controversy about the removal of XUL and XPCOM in Firefox, and several add-on authors refusing to update (eg. [DownThemAll](http://www.downthemall.net/the-likely-end-of-downthemall/)), I'm somewhat concerned:

In which ways will this affect the pentadactyl project? Are you going to migrate to the new WebExtension API? Would such a thing even be possible?

**Comment by tokiclover** [comment]
Clearly Mozilla want to make some bucks with the new hype with a
completely void corpse. Good luck to 'em then!

Well, I'll stick to the current ESR version for a year at least,
and keep it as long as possible untill I got something workable
elsewhere -- no Google backdoors(TM) wanted here... sorry.

I guess I will have to say good bye to this damn awesome add-on
along with NoScript, FashGOT et al.

BIG THANKS 'till the end of the end -- meaning when 38.8.0 will
hit the release table... and then a bit more 'till security-wise
no more functional nor practicable. So, yes, a year and a few
mounths.

Again, THANKS for this awesome add-on.

NOTE: Should we mourn...? let's laugh to pitfall of Mozilla instead!
Firefox was killed since v4 with that huge PR and plans to supposedly
catch up Chrome. -- Surely in the PITA GUI and smelly crufts reviewed
openly by BIG EYE/DISK.
I guess some no-brain and noob-skull from M$ would/will be happy to
get a fancy Mozilla backdoors(TM). Without me for sure.

**Comment by willsALMANJ** [comment]
It's amazing to see how many people mention Pentadactyl in the comments to that post -- way more than any other add-on.

Until we find out how the developers plan to address #79 though, these issues further out into the future are not worth worrying about.

**Comment by vyp** [comment]
@willsALMANJ If I'm not mistaken, @kmaglione is already on the AMO team, so I hope that won't be a problem.

But I think we should start to worry about OP's question now because otherwise the damage will be done by the time the deprecation happens. But perhaps it has already been done, and this is the end of pentadactyl as we know it? The first thing at least is to get an answer to @haasn's questions. +1

**Comment by willsALMANJ** [comment]
@vyp sorry, don't get me wrong, I think these announced changes have huge implications for Pentadactyl and I would love to hear from @kmaglione about them, but the signing issue has been looming for months with no word on if the developers will try to get it signed.

**Comment by vyp** [comment]
@willsALMANJ I agree of course, it's just that I think it's highly unlikely that it will *not* be signed. But yes, obviously it's a higher priority issue right now because 42 is only a few months away from becoming stable. But because Maglione is on AMO, it should be a quick and straightforward resolution I hope. (FTR, I do not support Mozilla being the only authority available to sign extensions in the first place, but that's another issue outside the scope of pentadactyl.)

But this issue has the potential to be the end of pentadactyl entirely, so I hope we get answers for both these questions some time soon, and I think we can worry about both simultaneously. :+1:

The post says:

> We have decided on an approximate timeline for the deprecation of XPCOM- and XUL-based add-ons.

But I may have missed it, did they actually post any dates? At least we know that #79 has to be resolved very soon. If we have a date for deprecation, it could help with prioritising this issue appropriately.

**Comment by willsALMANJ** [comment]
The dates are pretty fuzzy but here's what they say in the post:

* December 15th (Firefox 43 merges to release). The earliest release Electrolysis will be enabled on the release channel, and our current planned release.
* Six months past enabling Electrolysis on Release. The deprecation of CPOWs and compatibility shims will begin. 
* we have decided to deprecate add-ons that depend on XUL, XPCOM, and XBL. We don’t have a specific timeline for deprecation, but most likely it will take place within 12 to 18 months from now.

Both Electrolysis and the XUL deprecation will have major impacts on Pentadactyl

**Comment by tpruzina** [comment]
Guess we will just have to use unpatched firefox, screw security, I would personally refuse to use browser without pentadactyl.

**Comment by behrmann** [comment]
Currently wishes for the WebExtension API are being collected. I made a request, maybe a few votes on that could help 

https://webextensions.uservoice.com/forums/315663-webextension-api-ideas/suggestions/9455088-make-sure-add-ons-that-completely-change-the-ui-l
